File: app/view_models/prediction_point.rb
# frozen_string_literal: true

class PredictionPoint
  attr_reader \
    :generated_at,
    :target_date,
    :horizon,
    :predicted_nav,
    :lower_bound,
    :upper_bound,
    :confidence

  def initialize(
    generated_at:,
    target_date:,
    horizon:,
    predicted_nav:,
    lower_bound:,
    upper_bound:,
    confidence:
  )
    @generated_at = generated_at
    @target_date = target_date
    @horizon = horizon
    @predicted_nav = predicted_nav
    @lower_bound = lower_bound
    @upper_bound = upper_bound
    @confidence = confidence

    freeze
  end
end

File: mobile/src/models/PredictionPoint.ts
// models/PredictionPoint.ts

export type PredictionHorizon =
  | "1d"
  | "30d"
  | "90d";

export interface PredictionPoint {
  /**
   * When this prediction was generated.
   *
   * Used to determine which duplicate
   * prediction is the newest.
   */
  generated_at: string;

  /**
   * Forecast target date.
   */
  target_date: string;

  /**
   * Prediction horizon.
   */
  horizon: PredictionHorizon;

  /**
   * Predicted NAV.
   */
  predicted_nav: number | string;

  /**
   * Lower prediction bound.
   */
  lower_bound: number | string | null;

  /**
   * Upper prediction bound.
   */
  upper_bound: number | string | null;

  /**
   * Model confidence.
   */
  confidence: number | string | null;
}